packages/eslint-plugin/tests/rules/no-magic-numbers.test.ts

```diff
@@ -117,6 +117,104 @@ type Foo = {
       `,
       options: [{ ignoreTypeIndexes: true }],
     },
+    {
+      code: `
+type Foo =
+  | 1
+  | -2
+  | 3n
+  | -4n
+  | 5.6
+  | -7.8
+  | 0x0a
+  | -0xbc
+  | 1e2
+  | -3e4
+  | 5e-6
+  | -7e-8
+  | 1.1e2
+  | -3.1e4
+  | 5.1e-6
+  | -7.1e-8;
```

Comment:
[Testing] Continuing https://github.com/typescript-eslint/typescript-eslint/pull/9167#discussion_r1623415921: what I mean is, this test case right now does two things:

* 🍏 Check that a `type`'s union type's constituents are ignored properly
* 🍌 Check that all sorts of numeric literals (bigints, `0x`, `e`, ...) are ignored properly

...so, really, there should be a standalone test for 🍏 union type constituents. The 🍌 literal parsing doesn't need to be in that test.

```suggestion
```

From another perspective: this PR is only meant to fix 🍏 what the `ignore` option looks at. It's not changing 🍌 how the rule parses literals. So there's no need to add 🍌 literal parsing tests as part of this PR.

If you _really want_ to add more testing for 🍌 literal parsing (which is actually a good idea IMO, kind of a gap in testing at the moment!)_ then I'd be in favor of those. They'd be separate tests from the 🍏 type ones.

Reply:
So this one too should be split in multiple tests?

Reply:
Yes please 👍